components/scream/src/physics/share/physics_constants.hpp

```diff
@@ -108,8 +108,31 @@ struct Constants
   // = 3 Khairoutdinov and Kogan 2000
   static constexpr int IPARAM         = 3;
 
+  // Gases
+  static const std::map<std::string,Scalar> gas_mol_weights;
+  static Scalar get_gas_mol_weight(std::string gas_name);
 };
 
+// Gases
+// Define the molecular weight for each gas, which can then be
+// used to determine the volume mixing ratio for each gas.
+template <typename Scalar>
+const std::map<std::string,Scalar> Constants<Scalar>::gas_mol_weights = {
+  {"h2o", Scalar(Constants<Scalar>::MWH2O)},
+  {"co2", 44.0095},
+  {"o3" , 47.9982},
+  {"n2o", 44.0128},
+  {"co" , 28.0101},
+  {"ch4", 16.04246},
+  {"o2" , 31.998},
+  {"n2" , 28.0134}
+};
+
+template <typename Scalar>
+Scalar Constants<Scalar>::get_gas_mol_weight(std::string gas_name) {
```

Comment:
I don't think this function adds much. I mean, this function simply access the std map, so it is a host fcn only. On host, one could simply access the map by hand...

Reply:
Good point.  I agree this function doesn't add much.  My plan was to first develop a hook that may allow us to create a device friendly function later down the road.